### PR TITLE
Add native keyboard propositions lookalike

### PR DESCRIPTION
**Pull request by kelianClerc**
Add to grapes a view that looks like native keyboard propositions.

Naming and location of the component may be discussed though

https://user-images.githubusercontent.com/26220096/161007841-189e8f70-aef3-4f2d-963a-c91fccbf2139.mov

### CODE REVIEW

File: library/src/main/res/values/styles_selectors.xml [hunk 1]
```diff
@@ -25,9 +25,25 @@
         <item name="android:textSize">@dimen/tabCardViewTextSize</item>
         <item name="android:textColor">@color/selector_tab_cardview_text</item>
         <item name="android:fontFamily">@font/gt_america_bold</item>
-    </style>ew
+    </style>
+    ew
```
Comment (kelianClerc) [inline library/src/main/res/values/styles_selectors.xml:29]: oups

**Review by Tweener — CHANGES_REQUESTED** [review]
A few things to change here:
- The naming `proposition` is not correct in English for what we need.
- The way of building the TextViews could be improved to avoid a possible de-sync between the list of TextViews created and the number of configuration.items requested.
- The usage of a custom `OnItemClickedListener ` to avoid.

On another topic (already started on Slack): whether or not this view should be directly included in the `NumberKeyboard`, as an optional view.
I would prefer to enforce this `KeyboardPropositionsView` as part of the `NumberKeyboard` for the following reasons:
- Since this view is meant to be always linked to a keyboard, it would make sense that it is part of the `NumberKeyboard`, as optional, with the ability to configure it. From the point of view of our Design System (DS), we don't want to let the flexibility to the people using it to do whatever they want. On the contrary, we want to enforce this behavior of having suggestions specific to a keyboard.
In other words: `NumberKeyboard` remains the main atom and `KeyboardPropositionsView` is a molecule that is part of it. 
- We already have a view for a selector of items, outside of a keyboard, check: `PickerCardListView`. So for sure, this `KeyboardPropositionsView` will not be reused outside of the context of a keyboard.

File: library/src/main/java/com/spendesk/grapes/selectors/KeyboardPropositionsView.kt [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        val text: CharSequence
+    )
+
+    fun interface OnItemClickedListener {
+        fun onItemClicked(item: Item)
+    }
+
+    data class Configuration(
+        val items: List<Item>,
+        val maxItemsPerLine: Int = DEFAULT_MAX_ITEMS_PER_LINE
```
[the remark this answers is not in the document]
Comment (kelianClerc) [reply library/src/main/java/com/spendesk/grapes/selectors/KeyboardPropositionsView.kt:48]: Or I can simply remove that and let all items in the same line. It will be to the caller to make sure only what he wants is displayed

[the remark this answers is not in the document]
Comment (Tweener) [reply library/src/main/java/com/spendesk/grapes/selectors/KeyboardPropositionsView.kt:48]: >use val itemsToShow = configuration.items.take(configuration.maxItems)

Yeah that would be perfect. It's the responsibility of our DS to ensure it works properly no matter what the user asks for.

File: library/src/main/java/com/spendesk/grapes/NumberKeyboard.kt [hunk 3]
```diff
@@ -48,6 +50,14 @@ class NumberKeyboard : ConstraintLayout {
         val maximumDigits: Int = MAXIMUM_DIGITS_NONE
     )
 
+    data class Suggestions(
```
Comment (kelianClerc) [inline library/src/main/java/com/spendesk/grapes/NumberKeyboard.kt:53]: `Suggestion` witout s

File: library/src/main/java/com/spendesk/grapes/selectors/KeyboardSuggestionsView.kt [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    //endregion constructors
+
+    data class Item(
+        val id: String,
+        val text: CharSequence
+    )
+
+    data class Configuration(
+        val items: List<Item>,
+        val maxItems: Int = DEFAULT_MAX_ITEMS
```
Comment (Tweener) [inline library/src/main/java/com/spendesk/grapes/selectors/KeyboardSuggestionsView.kt:44]: Actually, I'm wondering if we really need this variable as configuration.
As the caller can pass a list of suggestions, the number of items is the max between that list size and DEFAULT_MAX_ITEMS.
Wdyt?

**Review by Tweener — CHANGES_REQUESTED** [review]
That much better with the suggestions within the keyboard view! 🔥 

A few things left to change and it'll be good!

File: library/src/main/java/com/spendesk/grapes/selectors/KeyboardSuggestionsView.kt [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    private val binding: SelectorKeyboardSuggestionsViewBinding = SelectorKeyboardSuggestionsViewBinding.inflate(LayoutInflater.from(context), this)
+
+    @StyleRes
+    private var itemStyle: Int = R.style.KeyboardSuggestionsViewTextStyle
+
+    @StyleRes
+    private var itemTextAppearance: Int = R.style.KeyboardSuggestionsViewTextAppearance
+    var onItemClicked: ((item: Item) -> Unit)? = null
+
+    fun updateItemTextAppearance(@StyleRes textAppearanceRes: Int) {
```
[the remark this answers is not in the document]
Comment (kelianClerc) [reply library/src/main/java/com/spendesk/grapes/selectors/KeyboardSuggestionsView.kt:57]: This class is internal to the DS because KeyboardSuggestionsView is now `internal`. I would prefer to keep it public for other mollecules using it to be able to customize it

File: library/src/main/java/com/spendesk/grapes/selectors/KeyboardSuggestionsView.kt (hunk 4 above)
Comment (kelianClerc) [reply library/src/main/java/com/spendesk/grapes/selectors/KeyboardSuggestionsView.kt:44]: You're right

File: library/src/main/java/com/spendesk/grapes/selectors/KeyboardSuggestionsView.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (Tweener) [reply library/src/main/java/com/spendesk/grapes/selectors/KeyboardSuggestionsView.kt:57]: (Open question, no need to change now) Shouldn't we do the other way in a Design System? Restrict the customization and if one day we need it, "open" it to external molecules/atoms?
Maybe @Scythe14 as some inputs on it as well?

File: library/src/main/java/com/spendesk/grapes/NumberKeyboard.kt [hunk 6]
```diff
@@ -123,6 +135,30 @@ class NumberKeyboard : ConstraintLayout {
         this.maximumDigits = configuration.maximumDigits
     }
 
+    fun updateSuggestions(suggestions: List<Suggestion>? = null) {
+        this.suggestions.clear()
+        suggestions?.let { this.suggestions.addAll(it) }
```
[the remark this answers is not in the document]
Comment (Tweener) [reply library/src/main/java/com/spendesk/grapes/NumberKeyboard.kt:140]: >When passing null suggestions are cleared and configuration is empty so suggestions are all removed.

Haha yes indeed, I misread the line and thought the clear was done in the `.let { ... }` 😅

File: library/src/main/java/com/spendesk/grapes/selectors/KeyboardSuggestionsView.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (kelianClerc) [reply library/src/main/java/com/spendesk/grapes/selectors/KeyboardSuggestionsView.kt:54]: Here it needs to be opened because the keyboard can be light or dark. 
But in most case, in my opinion, you're right, design system should be "as restrictive as possible" in term of style and appearance to make sure look and feel is consistent all over the app. 
Curious to hear @Scythe14 opinion too

**Review by Tweener — APPROVED** [review]
